File: CHFHEIST.py
T = int(input())

for _ in range(T):
    D, d, P, Q = map(int, input().strip().split())
    totalparts = D//d
    leftdays = D%d
    partialhold = d* totalparts * (2*P + (totalparts -1)*Q)
    partialhold //=2
    lefthold = leftdays * (P + totalparts*Q)
    print(partialhold + lefthold)

File: CHFHEIST.go
package main

import (
	"fmt"
)

func main() {
	var T, D, d, P, Q, partialhold, lefthold, totalparts, leftdays int
	fmt.Scanf("%d\n", &T)
	for i := 0; i < T; i++ {
		fmt.Scanf("%d %d %d %d\n", &D, &d, &P, &Q)
		totalparts = D / d
		// fmt.Println("totalparts", totalparts)
		leftdays = D % d
		//fmt.Println("leftdays", leftdays)
		partialhold = d * totalparts * (2*P + (totalparts-1)*Q)
		partialhold = partialhold / 2
		lefthold = leftdays * (P + totalparts*Q)
		//fmt.Println("partialhold", partialhold, "lefthold", lefthold)
		fmt.Println(partialhold + lefthold)
	}
}
